samples/network/manage-ip-address/Program.cs

```diff
@@ -180,81 +158,69 @@ public static async Task RunSample(TokenCredential credential)
-
-                // Get the associated public IP address for a virtual machine
-                Utilities.Log("Public IP address associated with the VM's primary NIC [After Update]");
-
-                publicIPDetail = (await publicIPAddresses.GetAsync(rgName, publicIPAddressName2)).Value;
-
-                Utilities.PrintIPAddress(publicIPDetail);
+                await networkInterfaceContainer.CreateOrUpdate(networkInterfaceName, networkInterfaceData).WaitForCompletionAsync();
+                Console.WriteLine("New public IP address associated with the VM's primary NIC");
 
                 // Remove public IP associated with the VM
-
-                Utilities.Log("Removing public IP address associated with the VM");
-
-                networkInterface = new NetworkInterface()
+                Console.WriteLine("Removing public IP address associated with the VM");
+                networkInterfaceData = new NetworkInterfaceData
                 {
                     Location = "eastus",
-                    IpConfigurations = new List<NetworkInterfaceIPConfiguration>()
+                    IpConfigurations =
                     {
                         new NetworkInterfaceIPConfiguration()
                         {
                             Name = "Primary",
                             Primary = true,
-                            Subnet = new Subnet() { Id = vnet.Subnets.First().Id },
+                            Subnet = new SubnetData() { Id = vnet.Data.Subnets.First().Id },
                             PrivateIPAllocationMethod = IPAllocationMethod.Dynamic,
                         }
                     }
                 };
 
-                await (await networkInterfaces.StartCreateOrUpdateAsync(rgName, networkInterfaceName, networkInterface)).WaitForCompletionAsync();
-
-                Utilities.Log("Removed public IP address associated with the VM");
+                await networkInterfaceContainer.CreateOrUpdate(networkInterfaceName, networkInterfaceData).WaitForCompletionAsync();
+                Console.WriteLine("Removed public IP address associated with the VM");
 
                 // Delete the public ip
-                Utilities.Log("Deleting the public IP address");
-                await (await publicIPAddresses.StartDeleteAsync(rgName, publicIPAddressName1)).WaitForCompletionAsync();
-                Utilities.Log("Deleted the public IP address");
+                Console.WriteLine("Deleting the public IP addresses");
+                await publicIPAddress.Delete().WaitForCompletionResponseAsync();
+                Console.WriteLine("Deleted the public IP addresses");
             }
             finally
             {
                 try
                 {
-                    await ResourceGroupHelper.DeleteResourceGroup(rgName);
+                    await resourceGroup.Delete().WaitForCompletionResponseAsync();
                 }
                 catch (NullReferenceException)
```

Comment:
think about why here need to catch NullReferenceException, can we remove it?

Reply:
This will be removed along with the removal of `try...catch`.